packages/eslint-plugin-pf-codemods/lib/rules/v5/dataList-updated-callback.js

```diff
@@ -0,0 +1,78 @@
+const { getPackageImports } = require("../../helpers");
+
+// https://github.com/patternfly/patternfly-react/pull/8723
+module.exports = {
+  meta: { fixable: "code" },
+  create: function (context) {
+    const imports = getPackageImports(context, "@patternfly/react-core").filter(
+      (specifier) => specifier.imported.name === "DataList"
+    );
+
+    return imports.length === 0
+      ? {}
+      : {
+          JSXOpeningElement(node) {
+            const onSelectDataListItemProp = node.attributes.find(
+              (attr) => attr.name.name === "onSelectDataListItem"
+            );
+
+            const { type, params, name } =
+              onSelectDataListItemProp?.value?.expression || {};
+
+            if (
+              imports.map((imp) => imp.local.name).includes(node.name.name) &&
+              ((type === "ArrowFunctionExpression" && params?.length === 1) ||
```

Comment:
consider class level functions too for reporting, regardless of if we can apply a fixer